integration_test/sql/subquery.exs

```diff
@@ -18,6 +18,7 @@ defmodule Ecto.Integration.SubQueryTest do
            TestRepo.all(from p in subquery(query), select: p)
   end
 
+  @tag :map_boolean_in_subquery
```

Comment:
This might be reflective of something I didn't quite understand properly about how to write an Ecto adapter. The test in question relied on being able to pass values from a subquery through placeholder values `?nnn` to the database engine.

The problem, for SQLite, is that this test uses a boolean value for the `public` field that comes through in a subquery. I did not see the dumper function get called for that value, so the boolean got passed through directly to the database engine.

SQLite does not have an intrinsic boolean data type, so I rely on the trick of replacing boolean with numeric 1 and 0. If that doesn't happen, I get an error running the query.

It's certainly possible that I misunderstood how the dumper should be invoked or maybe there's a bug in Ecto. Open to further discussion/guidance on how to proceed here.